attributes/default.rb

```diff
@@ -174,9 +178,9 @@
 default['chef_client']['log_rotation']['prerotate'] = nil
 default['chef_client']['log_rotation']['postrotate'] =  case node['chef_client']['init_style']
                                                         when 'systemd'
-                                                          'systemctl reload chef-client.service >/dev/null || :'
+                                                          "systemctl reload #{node['chef_client']['svc_name']}.service >/dev/null || :"
```

Comment:
Since these are derived attributes this won't actually work. You'll end up with the default value of chef-client being used even if you override. You basically end up with the old behavior unfortunately.